### PR TITLE
add trait `FilterByIdentifier`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Implement a filter trait select all types from a sequence those have a certain identifier.

### CODE REVIEW

File: src/libPMacc/include/particles/traits/FilterByIdentifier.hpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/* Copyright 2015-2017 Heiko Burau, Rene Widera
```
Comment (PrometheusPi) [inline src/libPMacc/include/particles/traits/FilterByIdentifier.hpp]: QUESTION:
@Heikman Please have look as well.

File: src/libPMacc/include/particles/traits/FilterByIdentifier.hpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+namespace PMacc
+{
+namespace particles
+{
+namespace traits
+{
+
+    /** Return a new sequence of particle species carrying flag.
```
Comment (PrometheusPi) [inline src/libPMacc/include/particles/traits/FilterByIdentifier.hpp]: OPTIONAL:
Return a new sequence of particles which carry the flag

Comment (Heikman) [inline src/libPMacc/include/particles/traits/FilterByIdentifier.hpp]: Please replace `flag` by `identifier`.

Comment (PrometheusPi) [inline src/libPMacc/include/particles/traits/FilterByIdentifier.hpp]: @Heikman: @psychocoderHPC fixed this line - I will merge

**Pull request merged**